src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -80,7 +107,53 @@ public override void Dispose()
 
         protected override bool OnSysCall(uint method)
         {
-            return InteropService.Invoke(this, method);
+            if (!services.TryGetValue(method, out InteropDescriptor descriptor))
+                return false;
+            if (!descriptor.AllowedTriggers.HasFlag(Trigger))
+                return false;
+            ExecutionContextState state = CurrentContext.GetState<ExecutionContextState>();
+            if (!state.CallFlags.HasFlag(descriptor.RequiredCallFlags))
+                return false;
+            if (!AddGas(descriptor.FixedPrice))
+                return false;
+            List<object> parameters = descriptor.Parameters.Length > 0
+                ? new List<object>()
+                : null;
+            foreach (var pd in descriptor.Parameters)
```

Comment:
I think that this could be slower than the previous version, that takes directly the values from the stack.

Reply:
It may be slower, but it is almost the same.

Reply:
But some syscalls do some `peeks` in order to validate or not the call, now it will consume it?

Reply:
Yes. Now all parameters will be consumed.